Source/WebCore/accessibility/AccessibilityNodeObject.cpp

```diff
@@ -823,17 +823,16 @@ bool AccessibilityNodeObject::isMultiSelectable() const
 
 bool AccessibilityNodeObject::isRequired() const
 {
-    // Explicit aria-required values should trump native required attributes.
+    Node* n = this->node();
+    if (is<HTMLFormControlElement>(n) && downcast<HTMLFormControlElement>(*n).isRequired())
+        return true;
```

Comment:
I know this code existed before, but since you're fixing it anyway let's make it a bit cleaner 🙂 

It can be written more simply like so:

```
auto* formControlElement = dynamicDowncast<HTMLFormControlElement>(node());
if (formControlElement && formControlElement->isRequired())
    return true;
```

`dynamicDowncast<T>` combines `is<T>` and `downcast<T>`, returning `nullptr` if the value is not a `T`.